Description: Use udev-in-Debian specific tools for the waiting script
 This is inspired from alsa-utils'
Author: Didier Raboud <didier@raboud.com>
Origin: vendor
Last-Update: 2010-12-01
--- a/usb_modeswitch.sh
+++ b/usb_modeswitch.sh
@@ -66,14 +66,7 @@
 		;;
 esac
 (
-count=120
-while [ $count != 0 ]; do
-	if [ ! -e "/usr/bin/tclsh" ]; then
-		sleep 1
-		count=$(($count - 1))
-	else
-		exec /usr/bin/tclsh /usr/sbin/usb_modeswitch_dispatcher "$@" 2>/dev/null &
-		exit 0
-	fi
-done
+	. /lib/udev/hotplug.functions
+	wait_for_file /usr/sbin/usb_modeswitch_dispatcher
+	exec /usr/sbin/usb_modeswitch_dispatcher "$@" 2>/dev/null &
 ) &
